### ISSUE TITLE
Pearls don't affect the pearled unless they're in a tileentity. They should affect while held, too

### ISSUE THREAD

**Issue by Lazersmoke** [DevotedMC/ExilePearl#30]
_From @GavJenks on March 3, 2018 23:37_

1) This is strictly a bug, since the instructions text on the config says "The radius around their pearl that an exiled player can't enter" which is definitely false as written.

2)  Even if rewritten, this is a feature we should have. yes, it may be annoying in some cases if a player is repeatedly teleported, but should be a feature anyway, because the plugin also allows for damage mode instead of teleporting, and it's pretty reasonable in that case. Even for teleporting, unless the pearler walks in exactly the direction of the victim, it shouldn't matter often.

Apologies if this is already a feature, but I looked through the config instructions and didn't see mention of it. Just "pearl_radius" alone

_Copied from original issue: CivClassic/ExilePearl#37_

**Comment by Lazersmoke** [comment]
@ProgrammerDan
Is there a reason why this exists?
https://github.com/CivClassic/ExilePearl/blob/32907821fefa6bfb3e0d218cf4933e77cbeeaad8/src/main/java/com/devotedmc/ExilePearl/core/PearlBoundaryTask.java#L149-L152
Is there something buggy that can happen or is it just to prevent weirdness like described above where you can get bumped back over and over?

**Comment by danquist** [comment]
I'm pretty sure Jon asked for this to be added after release to prevent the situation you're describing, and the config/docs were never updated. 

It could probably be a config option.

**Comment by ProgrammerDan** [comment]
Yeah the idea is it would be trivially easy to antagonistically use pearls in this case. Just load up a hot-potato player with pearls of a town you want to haze, run through, and enjoy the screams of frustration. This way, pearls had to be at least be marginally "fixed" in location; so yeah, you can still do _functionally_ the same, but at least you have to do some effort. 

So, not a bug; but, for sure not documented clearly.

**Comment by GavJenks** [comment]
That's kind of the ONLY usage for pearls themselves excluding people though, on a server with bastions.

If you have bastions, why wouldn't you turn off pearls doing anything, and just exclude from bastions (or damage in bastions)? Is that an option yet?

**Comment by ProgrammerDan** [comment]
I'm confused. You just have to put it in a chest. That's all...? They work as described then...?

**Comment by ProgrammerDan** [comment]
Oh, now I understand you. Late. You can just make the radius super tiny (0 or 1) and that's accomplished. Been available since day 1.
